src/Compile.hs

```diff
@@ -19,332 +19,543 @@ import           System.FilePath                ( replaceExtension
                                                 , dropFileName
                                                 , joinPath
                                                 )
+import           Explain.Location
 
 
+hpWrapLine :: Bool -> FilePath -> Int -> String -> String
+hpWrapLine coverage astPath line compiled = if coverage
+  then
+    "__hpLineWrap('" <> astPath <> "', " <> show line <> ", " <> compiled <> ")"
+  else compiled
+
+-- TODO: Pack all config into one param
```

Comment:
@aboeglin Please do this before merging, it starts to become very much overloaded with all the param threading.